sourced/ml/utils/spark.py

```diff
@@ -68,6 +74,17 @@ def create_spark(session_name,
     session.sparkContext.setLogLevel(spark_log_level)
     # Hide py4j verbose logging (It appears in travis mostly)
     logging.getLogger("py4j").setLevel(logging.WARNING)
+    if dep_zip:
+        tmp_dir = mkdtemp()
```

Comment:
Please use `~/.cache/source{d}/ml` instead and check if the archive exists. If it exists, fall through to the final `addPyFile`